### ISSUE TITLE
First-letter navigation in explorer tree views no longer works

### ISSUE THREAD

**Issue by Neurrone** [microsoft/vscode#179528]
Type: <b>Bug</b>

1. Press ctrl+shift+e to bring focus to the explorer window
2. Press the first letter of a file, e.g "p". 

Expected behaviour: focus moves to the next item starting with the character p, if one exists

Actual: nothing happens, even when a file like package.json exists.

This broke pretty recently, sometime within the last week.

VS Code version: Code - Insiders 1.78.0-insider (ea0e3e0d1fabf0622fbcacdc4fe1c7b73b34dc0b, 2023-04-07T05:20:50.322Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 1992)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.86GB (3.65GB free)|
|Process Argv|.|
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (31)</summary>

Extension|Author (truncated)|Version
---|---|---
astro-vscode|ast|0.28.2
vscode-eslint|dba|2.4.0
dendron|den|0.119.0
dendron-markdown-preview-enhanced|den|0.10.57
dendron-markdown-shortcuts|den|0.12.1
dendron-paste-image|den|1.1.1
dendron-snippet-maker|den|0.1.6
gitlens|eam|13.5.0
prettier-vscode|esb|9.10.4
rest-client|hum|0.25.1
rainbow-csv|mec|3.6.0
isort|ms-|2022.8.0
python|ms-|2023.6.0
vscode-pylance|ms-|2023.4.10
jupyter-keymap|ms-|1.1.0
remote-containers|ms-|0.289.0
remote-ssh|ms-|0.98.0
remote-ssh-edit|ms-|0.84.0
remote-wsl|ms-|0.78.0
vscode-remote-extensionpack|ms-|0.24.0
remote-explorer|ms-|0.4.0
remote-server|ms-|1.0.0
vsliveshare|ms-|1.0.5834
vsliveshare-audio|ms-|0.1.93
vsliveshare-pack|ms-|0.4.0
vscode-versionlens|pfl|1.5.0
rust-analyzer|rus|0.4.1469
svelte-vscode|sve|107.3.0
tauri-vscode|tau|0.2.6
shellcheck|tim|0.31.1
vscode-mermaid-editor|tom|0.18.1

</details>

**Comment by lramos15** [comment]
Even going back over a week it seems only the first key down used to work to focus a file and now it doesn't. I don't see any builds which each letter press would hop. Do you have an example? I think maybe this has something to do with `Ctrl+ F` taking over

**Comment by Neurrone** [comment]
I don't know what the term is but it used to do incremental navigation. For example, if I had both `package.json` and `pyproject.yml` in the same directory, typing "p" then "y" consecutively moves focus to pyproject.tml.

I'm using it with the NVDA screen reader, so I'm not sure if there's behavioural differences if VS Code detects that a screen reader is running.

/CC @isidorn

**Comment by OpatrilPeter** [comment]
I could reproduce this with all extensions disabled.

**Comment by isidorn** [comment]
fyi @meganrogge

**Comment by meganrogge** [comment]
@lramos15 can you take a look?

**Comment by gjsjohnmurray** [comment]
@lramos15 weirdly, when I build OSS from the 1.78.0 tag and try first-letter navigation on the explorer tree of the folder I open, it works. But when I repeat the test with the published 1.78.0 it doesn't. To rule out settings differences I pointed both of them at a fresh user-data-dir when launching.

Likewise when I use Insiders the navigation doesn't work, but if I build OSS from main it does work.

I am on Windows x64.

**Comment by lramos15** [comment]
> But when I repeat the test with the published 1.78.0 it doesn't. To rule out settings differences I pointed both of them at a fresh user-data-dir when launching.

Yeah it's weird, I'm still investigating here because there is quite a few reports. By bisecting built bits I've narrowed it down to the following commit range https://github.com/microsoft/vscode/compare/cbf35328c8357175b9cd747823f39ade987c3d3c...8f74fbfd1f2d8f6268a42df131726b218aafe511 to me it seems like the keybinding work might be to blame, but I'm not confident yet.

**Comment by lramos15** [comment]
Reopening to discuss for candidate process. Fix will be in tomorrow's insiders.

**Comment by roblourens** [comment]
I think we should take it as a candidate, but I think softDispatch returning null or NoMatchingKb is confusing. Here is another place where I think we do the wrong thing now

https://github.com/microsoft/vscode/blob/2c73d2651cc525b27fc697b820743098a204ec01/src/vs/workbench/services/hover/browser/hoverService.ts#L117

**Comment by starball5** [comment]
Likely related on Stack Overflow: [Why can't I type in VS Code 1.78.0's Explorer View to auto focus matching items?](https://stackoverflow.com/q/76199778/11107541)

**Comment by ulugbekna** [comment]
@roblourens 

> but I think softDispatch returning null or NoMatchingKb is confusing. Here is another place where I think we do the wrong thing now

I agree. I will refactor `softDispatch` to not return a null at all. For now, I made a small fix to possibly include in the recovery release - https://github.com/microsoft/vscode/pull/181887

**Comment by ulugbekna** [comment]
How to verify this is fixed: 

Focus on the file explorer, pressing a key should shift the focus to a file that starts with that key. For example, focusing on file explorer and pressing `g` should move the focus to the first file that starts with `g`.

**Comment by ahicks92** [comment]
Confirmed also that you can type for example "co" to get to core.  Looks like both cases are fixed.

Thanks.
